core/src/main/kotlin/br/com/concrete/yosef/api/property/spacing/MarginPropertyCommand.kt

```diff
@@ -0,0 +1,49 @@
+package br.com.concrete.yosef.api.property.spacing
+
+import android.view.View
+import android.widget.FrameLayout
+import android.widget.LinearLayout
+import br.com.concrete.yosef.api.property.DynamicPropertyCommand
+import br.com.concrete.yosef.dp
+import br.com.concrete.yosef.entity.DynamicProperty
+
+/**
+ * Command class that implements the [DynamicPropertyCommand] applying
+ * padding to a view view
+ *
+ * @see [View.setPadding]
+ */
+class MarginPropertyCommand : DynamicPropertyCommand {
+
+    companion object {
+        /**
+         * Name of the property that can be used in the json
+         */
+        const val MARGIN = "margin"
+    }
+
+    override fun apply(view: View, dynamicProperty: DynamicProperty) {
+        val layoutParams = view.layoutParams
+
+        val (left, top, right, bottom) = generateMarginList(dynamicProperty, view)
+
+        if (layoutParams is LinearLayout.LayoutParams) {
+            layoutParams.setMargins(left, top, right, bottom)
+        } else if (layoutParams is FrameLayout.LayoutParams) {
+            layoutParams.setMargins(left, top, right, bottom)
```

Comment:
I did not understand the need for this if expression here

Reply:
This is here because the default layout params type is ViewGroup.LayoutParams and it doesn't have setMargins, so i'm using kotlin smart cast to set it on the ViewGroups that we support.

Reply:
yeah, i got it but they are both doing the same method. Maybe the explicity cast would be better in this case or making these two expressions into one

Reply:
You can just check if the layout param is a https://developer.android.com/reference/android/view/ViewGroup.MarginLayoutParams.html.

Reply:
It will work with most view groups :)

Reply:
I think it's pretty self-explanatory, but what if we use a when? something like this:
```
when (layoutParams) {
    is LinearLayout.LayoutParams -> layoutParams.setMargins(left, top, right, bottom)
    is FrameLayout.LayoutParams -> layoutParams.setMargins(left, top, right, bottom)
}
```

Reply:
Very nice @mcassiano, i'll test it.

Reply:
done, it works